### PR TITLE
lpcxpresso55s69 post-hwmv2 fixup

### PR DESCRIPTION
**Pull request by rettichschnidi**
Based on #70218, I think this change should be incorporated.

However, I do not have access to the hardware and I do not yet understand hwmv2. Please review carefully.

### CODE REVIEW

File: boards/nxp/lpcxpresso55s69/Kconfig.defconfig [hunk 1]
```diff
@@ -30,7 +30,7 @@ DT_CHOSEN_Z_CODE_PARTITION := zephyr,code-partition
 
 config FLASH_LOAD_SIZE
 	default $(dt_chosen_reg_size_hex,$(DT_CHOSEN_Z_CODE_PARTITION))
-	depends on BOARD_LPCXPRESSO55S69_LPC55S69_CPU0 && TRUSTED_EXECUTION_SECURE
+	depends on BOARD_LPCXPRESSO55S69_LPC55S69_CPU0_NS && TRUSTED_EXECUTION_SECURE
```
Comment (nordicjm) [inline boards/nxp/lpcxpresso55s69/Kconfig.defconfig:33]: This is not true by looking at https://github.com/zephyrproject-rtos/zephyr/blob/086e144be7edab139c386d81e71bbd178bac259a/boards/nxp/lpcxpresso55s69/lpcxpresso55s69_lpc55s69_cpu0_ns_defconfig the ns defconfig disables `TRUSTED_EXECUTION_SECURE` and sets `TRUSTED_EXECUTION_NONSECURE`

**Comment by rettichschnidi** [comment]
Now that I read up a bit on hwmv2, I think that I got it wrong in https://github.com/zephyrproject-rtos/zephyr/pull/70218: https://github.com/zephyrproject-rtos/zephyr/pull/70218/files#r1534548559

File: tests/drivers/pwm/pwm_api/src/test_pwm.c [hunk 2]
```diff
@@ -81,7 +81,7 @@
 #define DEFAULT_PWM_PORT 2 /* TCC1/WO[2] on PA18 (D7) */
 #elif defined CONFIG_BOARD_MIMXRT685_EVK
 #define DEFAULT_PWM_PORT 7 /* D3 on Arduino connector J27 */
-#elif defined CONFIG_BOARD_LPCXPRESSO55S69_LPC55S69_CPU0
+#elif defined CONFIG_BOARD_LPCXPRESSO55S69_LPC55S69_CPU0_NS
```
Comment (rettichschnidi) [inline tests/drivers/pwm/pwm_api/src/test_pwm.c:84]: Is there anything that keeps this code from working on either, secure or non-secure builds? If not, I'd rather change this to the following:

```suggestion
#elif defined(CONFIG_BOARD_LPCXPRESSO55S69_LPC55S69_CPU0_NS) ||                                    \
	defined(CONFIG_BOARD_LPCXPRESSO55S69_LPC55S69_CPU0)
```

Comment (rettichschnidi) [reply tests/drivers/pwm/pwm_api/src/test_pwm.c]: Updated accordingly. I think this is correct now.

File: boards/nxp/lpcxpresso55s69/Kconfig.defconfig (hunk 1 above)
Comment (rettichschnidi) [reply boards/nxp/lpcxpresso55s69/Kconfig.defconfig]: done

**Review by nordicjm — APPROVED** [review]

**Review by decsny — APPROVED** [review]

**Review by mmahadevan108 — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Pull request merged**